### DISCUSSION TITLE
I am following a guide about accounts.js on the apollo blog but the @auth derective is not being recognized. I must be missing something obvious.

### DISCUSSION THREAD

**Discussion by tintin10q** [accounts-js/accounts#1098] [Q&A] [answered]
This is the guide I am following https://www.apollographql.com/blog/email-password-authentication-with-accounts-js-and-apollo-server/

I got the the Protecting our query part but when I put @auth in my typedefs it complains that it doesn't know about it.

The guide doesn't seem to even consider that this might happen.
https://imgur.com/Utfeamh

I must be missing something obvious please help acount.js looks realy nice but this is blocking me completly.

My code:
```
// index.js
require('dotenv').config();
// const typeDefs = require('./graphql/typedefs');
// const resolvers = require('./graphql/resolvers');
const subscriptions = require('./graphql/subscriptions');

const { ApolloServer, gql, makeExecutableSchema } = require('apollo-server');
const mongoose = require('mongoose');
const { Mongo } = require('@accounts/mongo');
const { mergeTypeDefs, mergeResolvers } = require('@graphql-toolkit/schema-merging');
const { AccountsServer } = require('@accounts/server');
const { AccountsPassword } = require('@accounts/password');
const { AccountsModule } = require('@accounts/graphql-api');
const {validateMessageBroker} = require('./messageBroker');
const {validateDataStore} = require('./dataManager');
// We connect mongoose to our local mongodb database
const PORT = process.env.PORT || 4000;
mongoose.connect(process.env.MONGOURL, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
});

const accountsMongo = new Mongo(mongoose.connection);

const typeDefs = gql`

    
    type Query {
        
        # This query will be protected so only authenticated users can access it taken from the guide it doesn't know @auth
        sensitiveInformation: String @auth
    }
`;


const resolvers = {
    Query: {
        sensitiveInformation: () => 'Sensitive info',
    },
};

validateDataStore().then(r => {console.log(r)})
validateMessageBroker().then(r => {console.log(r)})


const accountsPassword = new AccountsPassword({});

const accountsServer = new AccountsServer(
    {
        db: accountsMongo,
        // Replace this value with a strong secret
        tokenSecret: 'my-super-random-secret',
    },
    {
        password: accountsPassword,
    }
);

// We generate the accounts-js GraphQL module
const accountsGraphQL = AccountsModule.forRoot({ accountsServer });

// A new schema is created combining our schema and the accounts-js schema not merging correctly? 
const schema = makeExecutableSchema({
    subscriptions,
    typeDefs: mergeTypeDefs([accountsGraphQL.typeDefs, typeDefs ]),
    resolvers: mergeResolvers([accountsGraphQL.resolvers, resolvers]),
    schemaDirectives: {
        ...accountsGraphQL.schemaDirectives,
    },
});

const server = new ApolloServer({ schema, context: accountsGraphQL.context });

// The `listen` method launches a web server.
server.listen().then(({ url }) => {
    console.log(`🚀  Server ready at ${url}`);
});
```
How do I get to use @auth in my own shema?

**Comment by pradel** [comment, marked as answer]
Like this the code you are using looks good, is it only the IDE showing this error or do you also have it during at runtime?

**Reply by tintin10q** [reply]
The errors show up during runtime. My IDE also shows errors like @auth being red but the error logs I put are all from runtime. 

I start the app by doing `node -r esm index1.js` in the console. The `-r esm` is for the import statements [see here](https://stackoverflow.com/questions/47277887/node-experimental-modules-requested-module-does-not-provide-an-export-named) but I don't think that is the cause of the issue. 

Clearly I must be missing something big.

**Reply by tintin10q** [reply]
@pradel  you mentioned in the issue I made #1099 
> we use typescript for the examples so you will need to remove the types annotations in order to make it work.

I already understood that these are the errors I get after I remove the type annotations.

**Reply by pradel** [reply]
I am currently updating the example https://github.com/pradel/accounts-js-server-tutorial from the tutorial to make it work with the latest version of accounts-js looks like there was a problem. Will be ready a bit later today :)

**Reply by tintin10q** [reply]
Super! Thanks I am looking forward to looking at it 😄

**Reply by pradel** [reply]
I just updated the example and everything should be working now :) (you need to use v0.31.1)
https://github.com/pradel/accounts-js-server-tutorial
Let me know if you still have problems with the integration.

**Reply by tintin10q** [reply]
It worked! Thanks for helping me.
